openrtb-json/src/main/scala/com/powerspace/openrtb/json/bidrequest/OpenRtbImpressionSerde.scala

```diff
@@ -1,41 +1,58 @@
 package com.powerspace.openrtb.json.bidrequest
 
 import com.google.openrtb.BidRequest.Imp
-import com.google.openrtb._
 import com.powerspace.openrtb.json.EncoderProvider
+import com.powerspace.openrtb.json.bidrequest.OpenRtbBannerSerde.{OpenRtbBannerDecoder, OpenRtbBannerEncoder}
 import com.powerspace.openrtb.json.util.EncodingUtils
-import io.circe.Encoder
+import com.powerspace.openrtb.json.common.OpenRtbProtobufEnumEncoders
+import com.powerspace.openrtb.json.common.OpenRtbProtobufEnumDecoders
 import io.circe.generic.extras.Configuration
-import io.circe.generic.extras.semiauto.deriveEncoder
+import io.circe.{Decoder, Encoder}
+import io.circe.generic.extras.semiauto._
+
+/**
+  * OpenRTB Imp Encoder and Decoder
+  * @todo split up decoder and encoder
+  */
+object ImpressionLevelSerdes {
 
-object ImpressionLevelEncoders {
   import EncodingUtils._
+  import OpenRtbProtobufEnumEncoders._
+  import OpenRtbProtobufEnumDecoders._
 
-  import com.powerspace.openrtb.json.common.OpenRtbProtobufEnumEncoders._
+  private implicit val customConfig: Configuration = Configuration.default.withSnakeCaseMemberNames.withDefaults
 
-  import io.circe._
+  implicit val bannerEncoder: Encoder[Imp.Banner] = OpenRtbBannerEncoder.encoder
+  implicit val videoEncoder: Encoder[Imp.Video] = OpenRtbVideoSerde.encoder
+  implicit val audioEncoder: Encoder[Imp.Audio] = openRtbEncoder[Imp.Audio]
+  implicit val pmpEncoder: Encoder[Imp.Pmp] = OpenRtbPmpSerde.encoder
 
-  private implicit val customConfig: Configuration = Configuration.default.withSnakeCaseMemberNames
+  implicit val bannerDecoder: Decoder[Imp.Banner] = OpenRtbBannerDecoder.decoder
+  implicit val videoDecoder: Decoder[Imp.Video] = OpenRtbVideoSerde.decoder
+  implicit val audioDecoder: Decoder[Imp.Audio] = openRtbDecoder[Imp.Audio]
+  implicit val pmpDecoder: Decoder[Imp.Pmp] = OpenRtbPmpSerde.decoder
 
-  implicit val bannerEncoder = OpenRtbBannerSerde.encoder
-  implicit val videoEncoder = OpenRtbVideoSerde.encoder
-  implicit val audioEncoder: Encoder[Imp.Audio] = openrtbEncoder[Imp.Audio]
-  implicit val pmpEncoder = OpenRtbPmpSerde.encoder
 }
 
-/**
-  * OpenRTB Imp Serde
-  */
-object OpenRtbImpressionSerde {
+object OpenRtbImpressionSerde extends EncoderProvider[Imp] {
+
   import EncodingUtils._
-  private implicit val customConfig: Configuration = Configuration.default.withSnakeCaseMemberNames
+  import OpenRtbProtobufEnumDecoders._
```

Comment:
you import the decoders but not the encoder, why?

Reply:
`Imp` does not contain any enum so I can get rid of the decoders as well. Thanks